app/controllers/application_controller.rb

```diff
@@ -1,2 +1,13 @@
 class ApplicationController < ActionController::Base
+	before_action :set_current_user
+
+	def set_current_user
+		#finds user with sessions data and stores it if present
+		Current.user = User.where(:id => session[:user_id]).last if session[:user_id]
```

Comment:
is this required:
`if session[:user_id]`

Reply:
this action should only run when a user is signed in, thus seems necessary else it will run and might cause error.